index.js

```diff
@@ -101,17 +110,32 @@ module.exports = fp(
             }
         }
 
-        const stats = host
-            ? new Client({
-                  host,
-                  namespace,
-                  bufferSize,
-                  bufferFlushTimeout,
-                  udpDnsCache,
-                  udpDnsCacheTTL,
-                  onError: onError,
-              })
-            : clientMock();
+        let stats;
+
+        if (customDatsClient) {
```

Comment:
I am afraid that with this logic we lose the setup of the mocked client when we pass the custom instance.

Reply:
Never mind, there is no need to initialize a mock client when a custom one is passed.